### ISSUE TITLE
Migration elk 6.5

### ISSUE THREAD

**Issue by toog** [datasweet/kibana-datasweet-formula#34]
Hello

Have you planned to do the migration to Kibana 6.5 soon ?

**Comment by lfroment-datasweet** [comment]
Hi @toog 
Yes we will, but we still do not know if some new breaking change could cause some trouble. 
Stay tuned!

**Comment by toog** [comment]
Nice

That can help you : https://www.elastic.co/blog/kibana-plugin-api-changes-in-6-5

Thanks

**Comment by MatJohnson** [comment]
+1 Hoping for update for this fantastic plugin soon.

Thanks

**Comment by sincorecom** [comment]
+1 I love this plugin. But upgraded 6.5.1

Thanks!

**Comment by pchakour** [comment]
+1
@lfroment-datasweet 
I made a little patch to upgrade the plugin for kibana 6.5.
How can I share it with you ?

**Comment by sincorecom** [comment]
@pchakour  You can zip it up and attach it to this comment chain or Set up a Google link. One of us can then fork the project and apply your tweak so everyone can use it. 

I would love to use it!

**Comment by pchakour** [comment]
This is a zip which contain a git patch of my commit.

[0001-Upgrade-to-kibana-6.5.x.zip](https://github.com/datasweet/kibana-datasweet-formula/files/2659624/0001-Upgrade-to-kibana-6.5.x.zip)

To apply it, you can use the following command  : `git am < 0001-Upgrade-to-kibana-6.5.x.patch`

**Comment by sincorecom** [comment]
> This is a zip which contain a git patch of my commit.
> 
> [0001-Upgrade-to-kibana-6.5.x.zip](https://github.com/datasweet/kibana-datasweet-formula/files/2659624/0001-Upgrade-to-kibana-6.5.x.zip)
> 
> To apply it, you can use the following command : `git am < 0001-Upgrade-to-kibana-6.5.x.patch`

Perfect! I went ahead and made a branch and applied your patch worked perfect for me:
https://github.com/sincorecom/kibana-datasweet-formula

**Comment by lfroment-datasweet** [comment]
Hi all ! Sorry for such a long silence. We are quite busy on something that you could find useful when it is ready !
The release frequency of Elastic is hard to follow for us. Happy to hear that you patched that one successfully @pchakour . We will take a closer look in order to merge it.

**Comment by simpleuser99** [comment]
> > This is a zip which contain a git patch of my commit.
> > [0001-Upgrade-to-kibana-6.5.x.zip](https://github.com/datasweet/kibana-datasweet-formula/files/2659624/0001-Upgrade-to-kibana-6.5.x.zip)
> > To apply it, you can use the following command : `git am < 0001-Upgrade-to-kibana-6.5.x.patch`
> 
> Perfect! I went ahead and made a branch and applied your patch worked perfect for me:
> https://github.com/sincorecom/kibana-datasweet-formula

Hello!

**Comment by anomaliyao** [comment]
+1

**Comment by simpleuser99** [comment]
> Hi all ! Sorry for such a long silence. We are quite busy on something that you could find useful when it is ready !
> The release frequency of Elastic is hard to follow for us. Happy to hear that you patched that one successfully @pchakour . We will take a closer look in order to merge it.

When are you planning to release a plugin for kibana 6.5.x ?

**Comment by denemexuser** [comment]
Hi simpleuser99,
Could you please share your sweet.zip file ?
Thanks

**Comment by qakart** [comment]
I am looking for Kibana 6.5.1 is there anything I can give helping hand  to get it done

**Comment by lfroment-datasweet** [comment]
Hi all! Happy new year to everyone!
We are a small team and pretty busy on other projects right now. Following the latest version of Kibana is very tricky for us. We use 6.2 and lower for our clients. 6.5 implies too much breaking changes, we have no time for it yet. @pchakour provides a patch for 6.5. 

That is the best we can do know. Sorry folks, but we promise we will make it better in the next few months, with a brand new "gizmo" for you to take advantage of your ES clusters ;-) .

**Comment by SpecterHi** [comment]
@pchakour @sincorecom Thanks for your great job!
I wonder if it can be used it Kibana 6.5.4 since the official docker version is 6.5.4 and no 6.5.1

And Could you please release the distributable archive of the plugin?
I've tried to follow the development steps but failed...

Thanks!

**Comment by SpecterHi** [comment]
@pchakour Thank you! I will try again!

**Comment by SpecterHi** [comment]
@pchakour 
I followed the official guide, then apply your patch.
Since npm doesn't support the type "LINK", I used yarn instead.

Then I copy the whole folder to my kibana6.5.4 docker container, restarted it
Kibana started, and I saw "Datasweet Formula" in the Visualization.

But it doesn't calculate correctly... 
![image](https://user-images.githubusercontent.com/2437251/51987357-47a6d900-24dd-11e9-86bb-21ddc1f69421.png)

**Comment by pchakour** [comment]
I think it's an issue with the percentile metric. Use an other one and test again.
I will try to fix this later.

**Comment by SpecterHi** [comment]
> I think it's an issue with the percentile metric. Use an other one and test again.
> I will try to fix this later.

You are right! I just tried Max and Min, and it works!
Thank you so much!

**Comment by lfroment-datasweet** [comment]
Hi Guys,
we wanted to ask for some help if any of you wants to dive into that 6.5 / 6.6 support issue. 
We have no time available to do it in the near future, but we know that many are waiting for it. 

Is anyone willing to help us?

**Comment by toog** [comment]
We are on the same team pchakour and me. The patch he made is nice for elk 6.5.4 except an issue we discover when we use hidden metric.
I planned to make a PR for that.
So i think you can merge his patch to release.

**Comment by tcharlot-datasweet** [comment]
Hi, 
please give it a try to the last release (support for kibana v6.5.x to 6.7.x).
Regards
